src/scripts/main.js

```diff
@@ -2,4 +2,33 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+  const wallRectangle = wall.getBoundingClientRect();
+
+  let x = e.clientX - wallRectangle.left - spider.width + spider.width / 2;
+
+  let y = e.clientY - wallRectangle.top - spider.height + spider.height / 2;
```

Comment:
The calculation for `y` should ensure that the spider does not exceed the top boundary of the wall. The current calculation might cause the spider to exceed the top boundary if `e.clientY` is less than `wallRectangle.top + spider.height / 2`.